draft-ietf-taps-interface.md

```diff
@@ -1224,6 +1218,32 @@ received that reception has failed. Such conditions that irrevocably lead the
 the termination of the Connection are signaled using ConnectionError instead
 (see {{termination}}).
 
+## Receive Metadata
+
+Each Message may also contain metadata from protocols in the Protocol Stack; 
+which metadata is available is Protocol Stack dependent. The following metadata
+values are supported:
+
+### ECN {#receive-ecn}
+
+When available, Message metadata carries the value of the Explicit Congestion 
+Notification (ECN) field. This information can be used for logging and debugging 
+purposes, and for building applications which need access to information about 
+the transport internals for their own operation.
+
+### Early Data {#receive-early} 
+
+In some cases it may be valuable to know whether data was read as part of early 
+data streams. This is useful if applications need to treat early data separately, 
+e.g., if early data has different security properties than data sent after 
+connection establishment. In the case of TLS 1.3, client early data can be replayed 
+maliciously (see {{!I-D.ietf-tls-tls13}}). Thus, receivers may wish to perform additional 
+checks for early data to ensure it has not been replayed or restrict responses until 
+the handshake is finished. If TLS 1.3 is available and the recipient Message was 
+sent as part of early data, the corresponding metadata carries a flag indicating 
+as such. If early data is enabled, applications should check this metadata field 
+for Messages received during connection establishment and respond accordingly. 
+
```

Comment:
Better - although I wonder if it should be "Idempotent Early Data"?

Reply:
also +1 to "idempotent early data" (the former is implied by Early Data in a TLS context, but not necessarily in a transport context)

Reply:
That implies that the metadata signals whether or not the data is idempotent, which is something only applications can determine. I added some text clarifying this.